### DISCUSSION TITLE
Explanation for the automatic model conversion from homogeneous to heterogeneous

### DISCUSSION THREAD

**Discussion by enter802** [pyg-team/pytorch_geometric#9372] [Q&A]
Could someone please explain figure for the automatic conversion of a homogeneous model to heterogeneous model provided [here](https://pytorch-geometric.readthedocs.io/en/latest/notes/heterogeneous.html#automatically-converting-gnn-models)? Although I have implemented the process successfully, I cannot make sense of the figure. Will really appreciate if someone could shed some light.

![image](https://github.com/pyg-team/pytorch_geometric/assets/9440833/0e354b6a-7cd2-4dc4-983d-dbd2c8642eef)

**Comment by rusty1s** [comment]
Every GNN layer is duplicated for each edge type, and every node-level operation is duplicated for every node type. Edge-type specific embeddings from a GNN layer are then aggregated to each destination node type and form the final GNN embedding of the node type.

Imagine you have 2 node types (B, C) pointing to node type A. The pseudo code then becomes:
```
out_A = conv_B_A((x_B, x_A), edge_index_B_A) + conv_C_A((x_C, x_A), edge_index_C_A)
out_A = relu(out_A)

**Reply by enter802** [reply]
Sorry for the very late response but I was in ER for few days due to some health condition.

I understand what you have said. But I am having trouble connecting it to the figure. If you don't mind, could you please kindly provide a pseudo code for the figure?
